back/transcendence_django/back_game/monitor/channel/tournament_channel.py

```diff
@@ -0,0 +1,195 @@
+
+    def can_be_deleted(self) -> bool:
+        if len(self.users) == 0:
+            return True
+        return (
+            self.are_all_arenas_in_status_list([GameStatus(DEAD)])
+            and self.round_count == TOURNAMENT_MAX_ROUND + 1
+        )
+
+    def set_next_round(self):
+        if 1 <= self.round_count <= TOURNAMENT_MAX_ROUND:
+            self.__set_next_round_arenas()
+        self.round_count += 1
+        logger.info("Tournament round %s", self.round_count)
+
+    async def arena_loop(self, arena: Arena):
+        while self.round_count <= TOURNAMENT_MAX_ROUND and self.is_active:
+            logger.info("Arena %s loop", arena.id)
+            await super().arena_loop(arena)
+            await asyncio.sleep(ARENA_LOOP_INTERVAL)
+
+    async def next_round_loop(self):
+        while self.round_count <= TOURNAMENT_MAX_ROUND:
+            self.set_next_round()
+            await self.send_tournament_map()
+            await asyncio.sleep(WAIT_NEXT_ROUND_INTERVAL)
+            await self.send_assignations()
+            while not self.can_round_be_set():
+                if not self.is_active:
+                    return
+                logger.info("Waiting for next round")
+                await asyncio.sleep(NEXT_ROUND_LOOP_INTERVAL)
+
+    async def send_assignations(self):
+        assignations: dict[str, Any] = self.get_assignations()
+        await self.__send_update({ASSIGNATIONS: assignations})
+
+    async def send_tournament_map(self):
+        if self.tournament_map_sender is not None:
+            await self.tournament_map_sender()
+
+    def can_round_be_set(self):
+        return self.is_ready_to_start()
+
+    def get_tournament_map(self) -> Dict[str, RoundMapType | str | None]:
+        if self.winner:
+            logger.info(
+                "Tournament winner %s", self.winner.user_id
+            )
+        return {
+            ROUNDS_MAP: self.rounds_map,
+            TOURNAMENT_WINNER: self.winner.user_id if self.winner else None,
+        }
+
+    async def __send_update(self, data: dict[str, Any]):
+        if self.update_sender is not None:
+            logger.info("Sending assignations: %s", data)
+            await self.update_sender(data)
+
+    def __get_initial_rounds_map(self) -> RoundMapType:
+        rounds_map: RoundMapType = {}
+        for i in range(TOURNAMENT_MAX_ROUND):
+            round_players: RoundType = {}
+            arena_count = TOURNAMENT_ARENA_COUNT // 2**i
+            for j in range(arena_count):
+                round_players[str(j)] = [None for _ in range(self.players_specs[NB_PLAYERS])]
+            rounds_map[str(i + 1)] = round_players
+        return rounds_map
+
+    def __get_current_round_arenas(self, round_count: int) -> RoundMapType:
+        arena_count: int = len(self.rounds_map[str(round_count)])
+        round_arenas: RoundMapType = {}
+        user_ids = list(self.users.keys())
+        user_index = 0
+        for arena_id in range(arena_count):
```

Comment:
arena_id -> arena_index
Why use a dict with {"0": ..., "1": ..., ...} instead of a list? Serialization?